pkg/process/util/util.go

```diff
@@ -127,29 +127,40 @@ func GetSysRoot() string {
 	return "/sys"
 }
 
+// AllPidsProcs will return all pids under procRoot
+func AllPidsProcs(procRoot string) ([]int, error) {
+	f, err := os.Open(procRoot)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	dirs, err := f.Readdirnames(-1)
+	if err != nil {
+		return nil, err
+	}
+
+	pids := make([]int, 0, len(dirs))
+	for _, name := range dirs {
+		if pid, err := strconv.Atoi(name); err == nil {
```

Comment:
Does `Readdirnames` still return `.` and `..`?

I wonder if we should keep this test here:
```golang
if !f.IsDir() || f.Name() == "." || f.Name() == ".." {
	continue
}
```

Reply:
Readdirnames doesn't return `.` and `..` entries https://cs.opensource.google/go/go/+/master:src/os/dir_unix.go;l=107?q=readdir&ss=go%2Fgo:src%2Fos%2F

Reply:
Thanks for clarifying it!